packages/runtime/src/interpreter/interpreter.ts

```diff
@@ -8,16 +8,18 @@ import { DEFAULT_STACK_ELEM } from "../lib/constants";
 import { keyToBytes } from "../lib/parsing";
 import { Stack } from "../lib/stack";
 import { parser } from "../parser/parser";
-import type { Operator, SSCAttributesM, StackElem, StoreAccountI, TEALStack } from "../types";
+import type { ExecutionMode, Operator, SSCAttributesM, StackElem, StoreAccountI, TEALStack } from "../types";
 import { BIGINT0, BIGINT1, Label } from "./opcode-list";
 
 export class Interpreter {
   /**
    * Note: Interpreter operates on `ctx`, it doesn't operate on `store`.
    * All the functions query or update only a state copy from the interpreter, not the `runtime.store`.
    */
-
   readonly stack: TEALStack;
+  tealVersion: number; // LogicSigVersion
+  gas: number; // total gas cost of TEAL code
+  length: number; // total length of TEAL code
```

Comment:
How this length is expressed? In number of op codes? If yes, how about renaming to `opCodesNum`?

Reply:
It's actually the length of compiled teal code + arguments (for stateless) and 1kb for stateful execution mode. Not sure about how to calculate this, as we're not using the TEAL compiler.

Reply:
Added a new task to think about it: [https://www.pivotaltracker.com/story/show/176623588](https://www.pivotaltracker.com/story/show/176623588)